### PR TITLE
Update to follow proposed PayoutFunction spec

### PR DESCRIPTION
**Pull request by zpv**
+ Update payout_function_v0
+ Removes payout_function_v1
+ Add HyperbolaPayoutCurvePiece, PolynomialPayoutCurvePiece

### CODE REVIEW

File: packages/messaging/lib/messages/PayoutCurvePiece.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    buf: Buffer,
+  ): PolynomialPayoutCurvePiece | HyperbolaPayoutCurvePiece {
+    const reader = new BufferReader(buf);
+
+    const type = Number(reader.readBigSize());
+
+    switch (type) {
+      case MessageType.PolynomialPayoutCurvePiece:
+        return PolynomialPayoutCurvePiece.deserialize(buf);
+      case MessageType.HyperbolaPayoutCurvePiece:
```
Comment (matthewjablack) [inline packages/messaging/lib/messages/PayoutCurvePiece.ts:16]: Should this have a version number? Not sure what the answer is so added this to DLC Spec: https://github.com/discreetlogcontracts/dlcspecs/pull/144/files#r599245598

File: packages/messaging/lib/messages/PayoutFunction.ts [hunk 2]
```diff
@@ -78,131 +83,19 @@ export class PayoutFunctionV0 extends PayoutFunction implements IDlcMessage {
     const writer = new BufferWriter();
     writer.writeBigSize(this.type);
     writer.writeBigSize(this.length);
```
Comment (matthewjablack) [inline packages/messaging/lib/messages/PayoutFunction.ts:85]: Looks like length is getting written twice

Comment (matthewjablack) [reply packages/messaging/lib/messages/PayoutFunction.ts:85]: Maybe add a test for `PayoutFunction`

File: packages/messaging/lib/messages/PayoutCurvePiece.ts (hunk 1 above)
Comment (zpv) [reply packages/messaging/lib/messages/PayoutCurvePiece.ts:16]: yeah, isn’t specified in spec, so added a temp one for now

**Comment by matthewjablack** [comment]
One small change. Need to update `CoveredCall.spec.ts`

**Review by matthewjablack — CHANGES_REQUESTED** [review]
One small change. Need to update `CoveredCall.spec.ts` (still importing `PayoutFunctionV1`)

**Pull request merged**